src/tracer.js

```diff
@@ -228,6 +228,15 @@ export default class Tracer {
             ctx.parentId = null;
             ctx.flags = flags;
         } else {
+            if (!parent.samplingFinalized) {
+                if (this._sampler.isSampled(operationName, internalTags)) {
+                   parent._flags |= constants.SAMPLED_MASK
+                } else {
+                    parent._flags &= ~constants.SAMPLED_MASK
+                }
+                parent.finalizeSampling()
```

Comment:
don't like this. Parent context should be treated as immutable. Often it will be deserialized from the wire, so trying to modify it is misleading at best.